src/string_array.c

```diff
@@ -133,6 +134,51 @@ rcutils_string_array_cmp(
   return RCUTILS_RET_OK;
 }
 
+rcutils_ret_t
+rcutils_string_array_resize(
+  rcutils_string_array_t * string_array,
+  size_t new_size)
+{
+  RCUTILS_CHECK_FOR_NULL_WITH_MSG(
+    string_array, "string_array is null", return RCUTILS_RET_INVALID_ARGUMENT);
+
+  if (string_array->size == new_size) {
+    return RCUTILS_RET_OK;
+  }
+
+  if (0 == new_size) {
+    return rcutils_string_array_fini(string_array);
+  }
```

Comment:
I am not sure if this is intended behavior. `fini` also resets the allocators. Which means, IIUC, a user would be unable to resize to 0, followed by resizing to a positive number.

Reply:
It doesn't appear that `rcutils_string_array_fini` is resetting the allocator: https://github.com/ros2/rcutils/blob/6acf8d88906a9b304c7785c7b2ce1174a5ce4ecd/src/string_array.c#L63-L89

If that is a common practice, however, I'll remove the conditional and let it `reallocate` to zero.

Reply:
Well, now I am wondering whether `fini` should actually reset the allocator. I've seen mixed implementations here (e.g. char_array vs array_list). 
I guess this change is okay then as is.

Reply:
You've got me spooked, now. It's simple enough to not call `fini`. Obviously it's not the clear-cut optimization I thought it was.

Reply:
I did some research, and I think that calling `fini` really wasn't the right thing to do, so thanks for bringing it up. It actually pointed out that `init` is not consistently handling the implementation-specific behavior for what happens when you malloc or realloc to 0, so I'll open another PR to start that discussion.
I added a note to the docblock stating that even after resizing to 0, the `string_array` should still be `fini`'d.